### DISCUSSION TITLE
HTTPS for the public facing component of Akvorado

### DISCUSSION THREAD

**Discussion by trahman-hm** [akvorado/akvorado#1545] [Q&A]
### Disclaimer

- [X] I have read and understood the disclaimer.
- [X] I plan to implement the feature myself.

### Akvorado version

1.11.2

### Feature description

Would there be some way to force the use of HTTPS as the primary means of connecting to the Akvorado front end (tcp/8081) exposed to my network admins/engineers on my internal network?  Perhaps some way to shift the Traefik config to enable the use of HTTPS?

**Comment by vincentbernat** [comment]
Have a look at Traefik documentation: https://doc.traefik.io/traefik/https/acme/.

**Comment by dbray925** [comment]
I'm fronting ours with Apache, and passing the LDAP headers for user identification, and LDAP Group requirements. 

```
<Proxy *>
    Order deny,allow
    Allow from all
</Proxy>
ProxyPreserveHost On
<Location "/">
    ProxyPass "http://127.0.0.1:8082/"
    ProxyPassReverse "http://127.0.0.1:8082/"

    AuthName "Use Your LDAPN Credentials"
    AuthType Basic
    AuthBasicProvider ldap
    AuthLDAPInitialBindAsUser on
    AuthLDAPInitialBindPattern (.+) cn=$1,ou=Users,dc=domain,dc=lan
    AuthLDAPCompareAsUser on
    AuthLDAPURL "ldaps://openldap.domain.lan/ou=Users,dc=domain,dc=lan?uid?sub?(objectClass=*)"
    Require ldap-group cn=DevOps,ou=Groups,dc=domain,dc=lan
    RequestHeader set Remote-User %{AUTHENTICATE_uid}e
</Location>
```
